eo-runtime/src/main/java/org/eolang/PhTracedEnclosure.java

```diff
@@ -23,28 +23,31 @@
  */
 package org.eolang;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.function.Supplier;
 
 /**
  * Class to trace if the cage got into recursion during the dataization.
+ * NOT thread-safe.
  * @since 0.36
- * @todo #2836:60min Add a new parameter of recursion depth. This parameter
- *   should be set by user via pom.xml. We can make DATAIZING_CAGES a
- *   Map and count how many  times the cage was met.
  * @todo #2836:60min Make the class thread safe. It has private static
  *  field which can be accessed from differ thread and is not thread safe.
  *  Needs to synchronize this field.
  */
 @Versionized
 public final class PhTracedEnclosure implements Phi {
 
+    /**
+     * Name of property that responsible for keeping max depth.
+     */
+    public static final String MAX_CAGE_RECURSION_PROPERTY_NAME = "EO_MAX_CAGE_RECURSION";
```

Comment:
@levBagryansky I think the word "depth" should be present in the name of a variable and property itself.